x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -678,8 +676,11 @@ private SingleForecast forecast(Metadata metadata, DataStream stream, long forec
                 scaledTotalSize = totalSize;
             }
 
-            IndexMetadata writeIndex = metadata.index(stream.getWriteIndex());
+            if (numberNewIndices == 0) {
+                return null;
+            }
```

Comment:
I do not think this changes anything. The only reason we early terminate in other cases is to avoid NPE and division by zero.

Reply:
Right, this doesn't `numberNewIndices == 0` will not harm. This if statement at best is a tiny optimisation, by returning null, no empty `SingleForecast` is added to `singleForecast` in `forecast(...)` method that calls this method.

Do you want to remove this if statement from this PR?